src/types/complex-types/array.ts

```diff
@@ -149,7 +148,7 @@ export class ArrayType<S, T> extends ComplexType<S[], IObservableArray<T>> {
     }
 
     getDefaultSnapshot() {
-        return []
+        return undefined
```

Comment:
I think the only use case for `getDefaultSnapshot`  is `arrayFactory.create(/* no args*/)`, not skipping values from snapshots, so I don't think this change is needed.